src/components/data-management/DownloadData.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { useEffect } from 'react';
+import _ from 'lodash';
+import {
+  Menu, Tooltip, Dropdown, Button,
+} from 'antd';
+import { useSelector, useDispatch } from 'react-redux';
+import PropTypes from 'prop-types';
+import { saveAs } from 'file-saver';
+import pipelineStatus from '../../utils/pipelineStatusValues';
+import downloadData from '../../utils/data-management/downloadExperimentData';
+import downloadTypes from '../../utils/data-management/downloadTypes';
+import { exportQCParameters, filterQCParameters } from '../../utils/exportQCParameters';
+import { loadBackendStatus } from '../../redux/actions/backendStatus/index';
+
+const DownloadData = (props) => {
+  const {
+    activeProjectUuid,
+  } = props;
+  const dispatch = useDispatch();
+  const activeProject = useSelector((state) => state.projects[activeProjectUuid]);
+  const experimentSettings = useSelector((state) => state.experimentSettings);
+  const backendStatus = useSelector((state) => state.backendStatus);
+  const samples = useSelector((state) => state.samples);
+  const projects = useSelector((state) => state.projects);
+
+  useEffect(() => {
+    if (activeProject?.experiments?.length) {
+      dispatch(loadBackendStatus(activeProject?.experiments[0]));
+    }
+  }, [activeProject]);
+
+  const pipelineHasRun = (experimentId) => (
+    (backendStatus[experimentId]?.status.pipeline?.status === pipelineStatus.SUCCEEDED)
+    && activeProject?.experiments?.length
+  );
+  const gem2sHasRun = (experimentId) => (
+    (backendStatus[experimentId]?.status?.gem2s?.status === pipelineStatus.SUCCEEDED)
+   && activeProject?.experiments?.length
+  );
+
+  const allSamplesAnalysed = () => {
+    // Returns true only if there is at least one sample in the currently active
+    // project AND all samples in the project have been analysed.
+    const steps = Object.values(_.omit(experimentSettings?.processing, ['meta']));
+
+    return steps.length > 0
+      && activeProject?.samples?.length > 0
+      && activeProject?.samples?.every((s) => steps[0].hasOwnProperty(s));
+  };
```

Comment:
in this code, there is no point computing `steps` on line 44 if `activeProject?.samples?.length` is 0 and if `activeProject?.samples?.every((s) => steps[0].hasOwnProperty(s));`. Since checking for `activeProject?.samples?.length` is the "cheapest" in terms of complexity (it is basically O(1)), I would modify this `allSamplesAnalysed` function to first check for that, then return False if this is not the case, or if it is true, carry on with the other 2 checks.